lib/GithubUtils.js

```diff
@@ -119,4 +129,120 @@ export default class GithubUtils {
+     * @param {Array} [verifiedPRList] - The list of PR URLs which have passed QA.
+     * @param {Array} [deployBlockers] - The list of DeployBlocker URLs.
+     * @param {Array} [resolvedDeployBlockers] - The list of DeployBlockers URLs which have been resolved.
+     * @returns {Promise}
+     */
+    generateStagingDeployCashBody(
+        tag,
+        PRList,
+        verifiedPRList = [],
+        deployBlockers = [],
+        resolvedDeployBlockers = []
+    ) {
+        return this.generateVersionComparisonURL(`${GITHUB_OWNER}/${EXPENSIFY_CASH_REPO}`, tag, 'BUILD')
+            .then((comparisonURL) => {
+                const sortedPRList = _.sortBy(_.unique(PRList), URL => GithubUtils.getPullRequestNumberFromURL(URL));
+                // eslint-disable-next-line max-len
+                const sortedDeployBlockers = _.sortBy(_.unique(deployBlockers), URL => GithubUtils.getIssueOrPullRequestNumberFromURL(URL));
+
+                // Tag version and comparison URL
+                let issueBody = `**Release Version:** ${tag}\r\n`;
+                issueBody += `**Compare Changes:** ${comparisonURL}\r\n`;
+
+                // PR list
+                if (!_.isEmpty(PRList)) {
+                    issueBody += '**This release contains changes from the following pull requests:**\r\n';
+                    _.each(sortedPRList, (URL) => {
+                        issueBody += _.contains(verifiedPRList, URL) ? '- [x]' : '- [ ]';
+                        issueBody += ` ${URL}\r\n`;
+                    });
+                }
+
+                // Deploy blockers
+                if (!_.isEmpty(deployBlockers)) {
+                    issueBody += '\r\n**Deploy Blockers:**\r\n';
+                    _.each(sortedDeployBlockers, (URL) => {
+                        issueBody += _.contains(resolvedDeployBlockers, URL) ? '- [x]' : '- [ ]';
+                        issueBody += ` ${URL}\r\n`;
+                    });
+                }
+
+                return issueBody;
+            })
+            .catch(err => console.warn('Error generating comparison URL, continuing...', err));
+    }
+
+    /**
+     * Parse the pull request number from a URL.
+     *
+     * @param {String} URL
+     * @returns {String}
+     * @throws {Error} If the URL is not a valid Github Pull Request.
+     */
+    static getPullRequestNumberFromURL(URL) {
+        const matches = URL.match(/https:\/\/github\.com\/.*\/.*\/pull\/([0-9]+).*/);
+        if (!_.isArray(matches) || matches.length !== 2) {
+            throw new Error(`Provided URL ${URL} is not a Github Pull Request!`);
+        }
+        return matches[1];
+    }
+
+    /**
+     * Parse the issue number from a URL.
+     *
+     * @param {String} URL
+     * @returns {String}
+     * @throws {Error} If the URL is not a valid Github Issue.
+     */
+    static getIssueNumberFromURL(URL) {
+        const matches = URL.match(/https:\/\/github\.com\/.*\/.*\/issues\/([0-9]+).*/);
```

Comment:
NAB - Could you use `RegExp` here? I feel like it makes regex a little bit more legible without needing to escape everything.

Reply:
Okay, so this has been updated to use `RegExp`.

_However_, our understanding of how `RegExp` works compared to `/someRegexLiteral/`was slightly off. You _do_ need to escape regex characters such as `.`, except since you're using a regular string, the first forward-slash `\` that you would typically use to escape a character in a regex is "consumed" by the string.

That's because strings have their own set of special characters which need to be escaped, such as `\n`, and by default if it's not a special character then Javascript will just strip the `\` from the string. So in plain-old JS, `console.log('\.');` outputs `.`, _not_ `\.`.

So, in order to match a period literal using `RegExp`, you need to escape the forward slash first, like this: `RegExp('\\.')`. However, because `/` is not a special character in a string, you don't need to escape that. You only need to escape `/` using a regex literal because it's the character used to start/stop a regex literal.